filament/src/ShadowMap.cpp

```diff
@@ -1197,11 +1205,34 @@ backend::Viewport ShadowMap::getViewport() const noexcept {
     // We set a viewport with a 1-texel border for when we index outside the
     // texture. This can only happen for the directional light when "focus shadow casters is used".
     const uint32_t dim = mOptions->mapSize;
-    if (isDirectionalShadow()) {
-        return { 1, 1, dim - 2, dim - 2 };
-    } else {
-        return { 0, 0, dim, dim };
+    const uint16_t border = 1u;
+    return { border, border, dim - 2u * border, dim - 2u * border };
+}
+
+backend::Viewport ShadowMap::getScissor() const noexcept {
+    // We set a viewport with a 1-texel border for when we index outside the
+    // texture. This can only happen for the directional light when "focus shadow casters is used".
+    const uint32_t dim = mOptions->mapSize;
+    const uint16_t border = 1u;
+
+    switch (mShadowType) {
+        case ShadowType::DIRECTIONAL:
+            return { border, border, dim - 2u * border, dim - 2u * border };
+        case ShadowType::SPOT:
+        case ShadowType::POINT:
+        default:
+            return { 0, 0, dim, dim };
+    }
+}
+
+math::float4 ShadowMap::getViewportNormalized(ShadowMapInfo const& shadowMapInfo) const noexcept {
```

Comment:
Maybe we can think of name besides `viewport`? Since Filament treats a Viewport as {left, bottom, width, height} but this returns {left, bottom, right, top}.